spec/thread_safety_spec.rb

```diff
@@ -16,20 +14,65 @@
       Class.new do
         prepend MemoWise
 
-        def first_thread_id
-          Thread.pass # trigger a race condition even on MRI
-          Thread.current.object_id
+        def current_thread_id
+          Thread.pass              # trigger a race condition even on MRI
+          Thread.current.object_id # return different values in each thread
         end
-        memo_wise :first_thread_id
+        memo_wise :current_thread_id
       end
     end
 
     # Tip of the hat to @jeremyevans for finding and fixing a race condition
     # which caused accidental `nil` values to be returned under contended calls
     # to unmemoized zero-args methods.
     #   * See https://github.com/panorama-ed/memo_wise/pull/224
-    it "does not return accidental nil value to either thread" do
-      expect(thread_return_values).not_to include(nil)
+    context "when checking condition: are nil values accidentally returned?" do
+      let(:condition_to_check) do
+        ->(values) { values.any?(&:nil?) }
+      end
+
+      it "does not return accidental nil value to either thread" do
+        expect(thread_return_values).not_to include(nil)
+      end
+    end
+
+    # When memoizing non-deterministic methods, MemoWise does not impose any
+    # additional guarantees other than:
+    #   * Return values are one of the valid returns from the original method
+    #   * Return values differ between threads under contended calls
```

Comment:
This isn't a guarantee, right? Maybe we want _may_ differ or some other rephrasing?